### ISSUE TITLE
Is Pressflow 6 LTS still supported?

### ISSUE THREAD

**Issue by osopolar** [pressflow/6#130]
There is an issue for _pressflow6 version from drupal6.49_ #127 and a PR #128 for that, but since then  nothing happened. Will security updates be ported or would it be better to switch back from pressflow 6 to drupal 6 – is that even possible?

AFAICS there was no database update for switching from Drupal to Pressflow. The project description says:
> Each version of Pressflow is API-compatible with the same major Drupal version. For example, Pressflow 6 is compatible with all Drupal 6 modules.

If the site does not use some special Pressflow modules it should be possible to switch back, shouldn't it?

**Comment by omega8cc** [comment]
There is always some delay in syncing Pressflow 6 with Drupal 6 LTS, but you can always use our Pressflow 6 fork we maintain for many years, which also includes many improvements in comparison to vanilla Pressflow 6: https://github.com/omega8cc/pressflow6/releases

**Comment by osopolar** [comment]
> but you can always use our Pressflow 6 fork we maintain for many years, which also includes many improvements in comparison to vanilla Pressflow 6

Thanks @omega8cc, I didn't know about that. Is there any documentation on the differences between omega8cc/pressflow6 and pressflow/6? Is it still API compatible and easy to switch?

**Comment by osopolar** [comment]
I have done a diff between the last vanilla Pressflow 6 release (pressflow-6.47.127) and the corresponding one from omega8cc (https://github.com/omega8cc/pressflow6/releases/tag/pressflow-6.47.127), there where no differences. After that it looks like the _improvements_ are the changes from d6lts (https://github.com/d6lts/drupal/commits/6.x) to pressflow (https://github.com/omega8cc/pressflow6/compare/6.47.1...pressflow-plus), am I right? Or are there additional changes? 

Many thanks for maintaining pressflow and also your work on d6lts.

**Comment by pwolanin** [comment]
I have been trying to stay in sync with d6lts, but we could probably collaborate better and I'm not actually using this myself currently so it's hard to make it a priority.  That project has a pressflow branch that should match this though not exactly the same commits.

**Comment by omega8cc** [comment]
@osopolar We have started our fork with extensive set of patches recommended by experienced developers maintaining well known modules many years ago (2011). Some have been borrowed later from various D6 distros, because they newer got into vanilla nor Pressflow core. We haven't tracked changes introduced in d6lts, apart from merging releases and patches, but perhaps you have compared vanilla branch and not our customized branch? Here is where it started: https://groups.drupal.org/node/187209

We did the same with our D7 fork, btw: https://groups.drupal.org/node/210683

Yes, both Pressflow 6 and Drupal 7 forks we manage are fully compatible with vanilla core, no API changes, just fixes and improvements.

**Comment by omega8cc** [comment]
@osopolar Ah, you were comparing the same tags/points in Pressflow and our fork repo. They are obviously identical, because we merge everything, including tags. You should compare our main branch with Pressflow main branch instead.

**Comment by dsnopek** [comment]
Regarding what @pwolanin wrote:

> I have been trying to stay in sync with d6lts, but we could probably collaborate better and I'm not actually using this myself currently so it's hard to make it a priority. That project has a pressflow branch that should match this though not exactly the same commits.

We do merge all the D6LTS changes into our own Pressflow fork, but we've diverged a little from upstream Pressflow, and haven't been great about making PR's to merge these back into upstream. Here's the latest branch:

https://github.com/d6lts/pressflow/tree/d-652

I'll try to find some time to figure out our differences with upstream Pressflow and try to get making a PR part of our standard process.

**Comment by dsnopek** [comment]
Here's PR that would sync us up:

https://github.com/pressflow/6/pull/131

**Comment by osopolar** [comment]
@omega8cc Do you still maintain Pressflow 6 (https://github.com/omega8cc/pressflow6/issues)?

**Comment by omega8cc** [comment]
@osopolar Yes, it's slower than usual, but we will update our fork to sync it with LTS this week.

**Comment by omega8cc** [comment]
Yes, our Pressflow fork 6.56.2 has been released. We will make sure it gets any further updates from D6LTS faster.

**Comment by osopolar** [comment]
@omega8cc You said  on 4 Jun 2019:
> Yes, both Pressflow 6 and Drupal 7 forks we manage are fully compatible with vanilla core, no API changes, just fixes and improvements.

I am not sure, what compatibility with vanilla core means, but if you mean exactly the same behavior, it does not. We run into one problem, where an editor disabled a menu item (using og_menu module) can't enable it again due to https://github.com/omega8cc/pressflow6/commit/0a30619066ec47461401f6aa6ad19138b98f2547

**Comment by omega8cc** [comment]
>@omega8cc You said  on 4 Jun 2019: ...

@osopolar By compatibility we mean API compatibility, not the same behavior, for obvious reasons — many improvements and changes do change behavior — that’s the point of any performance improvement for example. It obviously doesn’t guarantee zero bugs or side effects.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>
